com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/AndroidBundler.java

```diff
@@ -782,16 +783,21 @@ private static File createUniversalApks(Project project, File aab, File outDir,
             args.add("--ks-key-alias"); args.add(keystoreAlias);
             args.add("--key-pass"); args.add("file:" + keyPasswordFile);
 
-            Result res = exec(args);
-            if (res.ret != 0) {
-                String msg = new String(res.stdOutErr);
-                throw new IOException(msg);
-            }
-            BundleHelper.throwIfCanceled(canceled);
-            return new File(apksPath);
+            res = exec(args);
         } catch (Exception e) {
             throw new CompileExceptionError("Failed creating universal APK", e);
         }
+        BundleHelper.throwIfCanceled(canceled);
+        if (res.ret == 0) {
+            return new File(apksPath);
+        }
+        String msg = new String(res.stdOutErr);
+        if (msg.contains("java.lang.ArithmeticException: integer overflow")) {
```

Comment:
When the APK is too large the output from bundletool.jar will result in an ArithmeticException:

```
[BT:1.14.0] Error: integer overflow 
java.lang.ArithmeticException: integer overflow
	at java.base/java.lang.Math.toIntExact(Math.java:1135)
	at com.android.zipflinger.Location.compareTo(Location.java:71)
	at com.android.zipflinger.Location.compareTo(Location.java:21)
	at java.base/java.util.ComparableTimSort.countRunAndMakeAscending(ComparableTimSort.java:325)
	at java.base/java.util.ComparableTimSort.sort(ComparableTimSort.java:188)
	at java.base/java.util.Arrays.sort(Arrays.java:1107)
	at java.base/java.util.Arrays.sort(Arrays.java:1301)
	at java.base/java.util.ArrayList.sort(ArrayList.java:1721)
	at java.base/java.util.Collections.sort(Collections.java:145)
```